csp/middleware.py

```diff
@@ -23,6 +26,14 @@ class CSPMiddleware(MiddlewareMixin):
     See http://www.w3.org/TR/CSP/
 
     """
+    def _make_nonce(self, request, length=16):
+        # Return cached value if this has been executed already
+        if not getattr(request, '_csp_nonce', None):
+            request._csp_nonce = get_random_string(length)
```

Comment:
Ah, I see. I didn't have the context processor in my mind when I looked at this.

Can you update the comment to make it a bit more clear? Also, I'm wondering if this method should better be called `_get_nonce` since that's more effectively what it does, it's returning a nonce and `make` kinda implies a re-generation imho.